### PR TITLE
Converted Certificate Sample to TypeScript

### PR DESCRIPTION
**Pull request by Robbie-Microsoft**
This is the first time we'll have an E2E test running on a TypeScript sample. They all currently run on JS samples.

### CODE REVIEW

**Review by bgavrilMS — APPROVED** [review]

**Comment by gladjohn** [comment]
is there a bug associated with this? Why are we creating a ts sample. Any context may help with review

File: samples/msal-node-samples/cachePlugin.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import fs from "fs";
+import { TokenCacheContext, ICachePlugin } from "@azure/msal-node";
+
+const cachePluginFunctionality = (cacheLocation: string): ICachePlugin => {
+    const beforeCacheAccess = (
+        cacheContext: TokenCacheContext
+    ): Promise<void> => {
+        return new Promise((resolve, reject) => {
+            if (fs.existsSync(cacheLocation)) {
```
Comment (bgavrilMS) [inline samples/msal-node-samples/cachePlugin.ts]: I don't think a plaintext file is appropriate caching location for service tokens.

For app to app, just use memory. Bonus points for using an in-memory cache with eviction.

Comment (bgavrilMS) [reply samples/msal-node-samples/cachePlugin.ts]: Also, you should not cache ALL your tokens in the same location. Token cache lookup in MSAL NodeJS is O(n) (at least). 

For S2S, you need to partition the cache. Dogan wrote some helpers that shows how to do this for user, see distributedcacheplugin. https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/caching.md#web-apps

Otherwise a multi-tenant app caching 1M tokens will just not work.

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
Do not use flat cache in confidntial client.

File: samples/msal-node-samples/cachePlugin.ts (hunk 1 above)
Comment (Robbie-Microsoft) [reply samples/msal-node-samples/cachePlugin.ts]: This is just a sample. All other JS samples use a flat cache like this, and I believe the point is so that devs can see how the tokens are stored in the cache. I can remove if you'd like.

Comment (bgavrilMS) [reply samples/msal-node-samples/cachePlugin.ts]: Caching is good. Flat file cache not so much.

File: samples/msal-node-samples/client-credentials-with-cert-from-key-vault/tsconfig.json [hunk 2]
```diff
@@ -1,3 +1,23 @@
 {
-    "extends": "../tsconfig.json"
```
Comment (tnorling) [inline samples/msal-node-samples/client-credentials-with-cert-from-key-vault/tsconfig.json:2]: Leave the extends and add specific settings you want to change here

File: samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs [hunk 3]
```diff
@@ -6,4 +6,5 @@
 module.exports = {
     displayName: "Client Credentials with Certificate from Key Vault",
     preset: "../../e2eTestUtils/jest-puppeteer-utils/jest-preset-no-setup.js",
+    extensionsToTreatAsEsm: [".ts", ".tsx"],
```
Comment (tnorling) [inline samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs:9]: Why is this needed?

Comment (Robbie-Microsoft) [reply samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs]: It's not, I deleted it

Comment (Robbie-Microsoft) [reply samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs:9]: I was getting some build errors without this

Comment (Robbie-Microsoft) [reply samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs:9]: https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1385457&view=logs&j=e2a201cd-fff7-5f83-ed5f-a5e7e4e44190&t=b5e84f64-0e7c-5c3b-7eb2-5014792c7f1e&l=80

Comment (tnorling) [reply samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs:9]: Is this an old run? It's referencing `jest.config.js` being treated as ESM but the file is, correctly, named `jest.config.cjs` now

Comment (Robbie-Microsoft) [reply samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs:9]: It is an old run. I receive that error when the file ends with ".js" instead of ".cjs". I can Google the difference I suppose, but for the purpose of discussion here, what is the ;TLDR difference between the two?

I can try keeping the file as ".cjs" and removing that line you asked the question about, and see if I still receive the build error.

Comment (tnorling) [reply samples/msal-node-samples/client-credentials-with-cert-from-key-vault/jest.config.cjs:9]: Different module formats. If you've specified `type: "module"` in the package.json, `.js` file extensions are treated as ESM. Without the `type` specified `.js` extensions are treated as CJS. To explicitly use CJS or ESM use `.cjs` or `.mjs` respectively. In this case the config is written using CJS syntax so it either needs to use `.cjs` or remove `type: "module"` from the package.json

**Review by bgavrilMS — APPROVED** [review]
